pkg/apiserver/rest/dualwriter.go

```diff
@@ -61,21 +58,22 @@ type LegacyStorage interface {
 // - rest.Updater
 // - rest.GracefulDeleter
 // - rest.CollectionDeleter
-type DualWriter struct {
+
+type DualWriter interface {
 	Storage
-	Legacy LegacyStorage
+	LegacyStorage
 }
 
+type DualWrite struct{ DualWriter }
+
 var errDualWriterCreaterMissing = errors.New("legacy storage rest.Creater is missing")
 var errDualWriterListerMissing = errors.New("legacy storage rest.Lister is missing")
 var errDualWriterDeleterMissing = errors.New("legacy storage rest.GracefulDeleter is missing")
 var errDualWriterCollectionDeleterMissing = errors.New("legacy storage rest.CollectionDeleter is missing")
 var errDualWriterUpdaterMissing = errors.New("legacy storage rest.Updater is missing")
 
-type DualWriterMode int
-
```

Comment:
What's the rational for removing the type ?

Reply:
I found it a bit confusing among more dual writer related definitions, such as `dualwriter`, `dualwrite` and `dualwriterModeX`. It's also only used to define the modes, so I thought it would be more clear to just assign modes to iota types.

Reply:
An argument to keep is that we could then replace the [existing boolean flag ](https://github.com/grafana/grafana/blob/main/pkg/registry/apis/playlist/register.go#L82)with this enum.

And add comments that explain each of the modes so I don't have to keep looking them up 😛

Reply:
ah, that's a good idea and already a preparation for https://github.com/grafana/search-and-storage-team/issues/15 :)

(and yes, will add comments for all the modes!)